.github/workflows/main.yml

```diff
@@ -88,37 +83,113 @@ jobs:
 
     - name: Build
       run: |
-        if [ "${{matrix.mode}}" == "test" ]; then
            make debug
-        else
-           make release
-        fi
 
     - name: Test
-      if: matrix.mode == 'test'
       run: |
         make test/unittest-d
 
-    - name: Regression
-      timeout-minutes: 180
-      if: matrix.mode == 'regression'
+  linux-regression:
+    name: "Regression ${{ matrix.compiler }} ${{matrix.shard}} (Linux)"
+
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        num_shards: [3]
+        shard: [0, 1, 2]
+        compiler:
+        - gcc
+        - clang
+
+    env:
+      COMPILER: ${{ matrix.compiler }}
+
+    steps:
+
+    - name: Cancel previous
+      uses: styfle/cancel-workflow-action@0.11.0
+      with:
+        access_token: ${{ github.token }}
+
+    - name: Install dependencies
       run: |
-        make test/regression
-        git status
+        sudo apt-get update -qq
```

Comment:
use https://github.com/marketplace/actions/cache-apt-packages to speed thing up or build a local container image in another job marked as a dependencies of all the shards?